crates/wasmtime/src/runtime/vm/traphandlers.rs

```diff
@@ -387,7 +430,7 @@ pub unsafe fn catch_traps<T, F>(
     store: &mut StoreContextMut<'_, T>,
     old_state: &mut EntryStoreContext,
     mut closure: F,
-) -> Result<(), Box<Trap>>
+) -> Result<()>
```

Comment:
With the changes here and to the `match` below, could you run the `call.rs` benchmark before/after this PR with the `^sync.*no-hook.*host-to-wasm.* typed - nop$` regex? If there's no real change, yay! If there's a regression, mind spot-checking to see if anything jumps out?

Reply:
That's reasonable enough for now IMO. I suspect we might be able to recover that by shuffling around things, maybe adding `#[cold]`, tweaking slow paths, etc. That being said it's still solidly in the realm of "this shouldn't matter and it's still fast", so I think it's fine to defer such theoretical tweaks to a future point in time.